deep_field_metadetect/gaussmom/gaussmom_core.py

```diff
@@ -0,0 +1,120 @@
+from typing import NamedTuple
+
+import jax
+import jax.numpy as jnp
+import ngmix
+
+
+@jax.tree_util.register_pytree_node_class
+class GaussMomObs(NamedTuple):
+    u: jax.Array
+    v: jax.Array
+    image: jax.Array
+    area: float
+    pixelwise_wgt: jax.Array
+
+    def tree_flatten(self):
+        return (
+            self.u,
+            self.v,
+            self.image,
+            self.area,
+            self.pixelwise_wgt,
+        ), None
+
+    @classmethod
+    def tree_unflatten(cls, aux_data, children):
+        return cls(*children)
+
+
+@jax.tree_util.register_pytree_node_class
+class GaussMomData(NamedTuple):
```

Comment:
We should be pre-fixing classes and functions with `_` if they are not meant to be used by users.